builtin/providers/aws/resource_aws_s3_bucket.go

```diff
@@ -118,3 +156,96 @@ func resourceAwsS3BucketDelete(d *schema.ResourceData, meta interface{}) error {
 	}
 	return nil
 }
+
+func resourceAwsS3BucketWebsiteUpdate(s3conn *s3.S3, d *schema.ResourceData) error {
+	if !d.HasChange("website") {
+		return nil
+	}
+
+	ws := d.Get("website").([]interface{})
+
+	if len(ws) == 1 {
+		w := ws[0].(map[string]interface{})
+		return resourceAwsS3BucketWebsitePut(s3conn, d, w)
+	} else if len(ws) == 0 {
+		return resourceAwsS3BucketWebsiteDelete(s3conn, d)
+	} else {
+		return fmt.Errorf("Cannot specify more than one website.")
+	}
+}
+
+func resourceAwsS3BucketWebsitePut(s3conn *s3.S3, d *schema.ResourceData, website map[string]interface{}) error {
+	bucket := d.Get("bucket").(string)
+
+	indexDocument := website["index_document"].(string)
+	errorDocument := website["error_document"].(string)
+
+	websiteConfiguration := &s3.WebsiteConfiguration{}
+
+	websiteConfiguration.IndexDocument = &s3.IndexDocument{Suffix: aws.String(indexDocument)}
+
+	if errorDocument != "" {
+		websiteConfiguration.ErrorDocument = &s3.ErrorDocument{Key: aws.String(errorDocument)}
+	}
+
+	putInput := &s3.PutBucketWebsiteInput{
+		Bucket:               aws.String(bucket),
+		WebsiteConfiguration: websiteConfiguration,
+	}
+
+	log.Printf("[DEBUG] S3 put bucket website: %s", putInput)
+
+	_, err := s3conn.PutBucketWebsite(putInput)
+	if err != nil {
+		return fmt.Errorf("Error putting S3 website: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsS3BucketWebsiteDelete(s3conn *s3.S3, d *schema.ResourceData) error {
+	bucket := d.Get("bucket").(string)
+	deleteInput := &s3.DeleteBucketWebsiteInput{Bucket: aws.String(bucket)}
+
+	log.Printf("[DEBUG] S3 delete bucket website: %s", deleteInput)
+
+	_, err := s3conn.DeleteBucketWebsite(deleteInput)
+	if err != nil {
+		return fmt.Errorf("Error deleting S3 website: %s", err)
+	}
+
+	return nil
+}
+
+func websiteEndpoint(s3conn *s3.S3, d *schema.ResourceData) (endpoint string, err error) {
+	// If the bucket doesn't have a website configuration, return an empty
+	// endpoint
+	if len(d.Get("website").([]interface{})) == 0 {
+		return
+	}
+
+	bucket := d.Get("bucket").(string)
+
+	// Lookup the region for this bucket
+	location, err := s3conn.GetBucketLocation(
+		&s3.GetBucketLocationInput{
+			Bucket: aws.String(bucket),
+		},
+	)
+	if err != nil {
+		return
+	}
+	var region string
+	if location.LocationConstraint != nil {
+		region = *location.LocationConstraint
+	}
+
+	// Default to us-east-1 if the bucket doesn't have a region
+	if region == "" {
+		region = "us-east-1"
+	}
+
+	endpoint = fmt.Sprintf("%s.s3-website-%s.amazonaws.com", bucket, region)
+
+	return
```

Comment:
This is just a personal preference, but I prefer explicit returns when possible. I don't think that's a requirement, but I believe most of TF is consistent there.